### PR TITLE
feat: Field existence and type check for SpringCriteria INTELLIJ-51, INTELLIJ-53

### PR DESCRIPTION
**Pull request by himanshusinghs**
### Checklist

- [ ] New tests and/or benchmarks are included.
- [ ] Documentation is changed or added.
- [ ] Changelog is updated accordingly.
- [ ] I have signed the MongoDB Contributor License Agreement (https://www.mongodb.com/legal/contributor-agreement).

### CODE REVIEW

File: packages/jetbrains-plugin/src/main/kotlin/com/mongodb/jbplugin/inspections/AbstractMongoDbInspectionBridge.kt [hunk 1]
```diff
@@ -28,6 +30,11 @@ abstract class AbstractMongoDbInspectionBridge(
             "QueryForDialect${dialect.javaClass.name}",
         )
 
+    /**
+     * Ktlint complains about buildVisitor being longer than 50 lines but because it is just an object implementation
+     * it should be fine to keep it like this to favor readability
+     */
+    @Suppress("ktlint")
```
Comment (kmruiz) [inline packages/jetbrains-plugin/src/main/kotlin/com/mongodb/jbplugin/inspections/AbstractMongoDbInspectionBridge.kt:37]: I agree that suppressing here is fine. However, I would suggest suppressing only the issue itself, in this case, @Suppress("TOO_LONG_FUNCTION").

File: packages/jetbrains-plugin/src/main/kotlin/com/mongodb/jbplugin/inspections/AbstractMongoDbInspectionBridge.kt [hunk 2]
```diff
@@ -65,21 +72,34 @@ abstract class AbstractMongoDbInspectionBridge(
...
                         }
                     }
                 }
             }
+
+            private fun queryWithCollectionReference(query: Node<PsiElement>, virtualFile: VirtualFile) =
+                MongoDbVirtualFileDataSourceProvider().getDatabase(
+                    virtualFile,
+                )?.let { database ->
+                    query.copy { component ->
```
Comment (kmruiz) [inline packages/jetbrains-plugin/src/main/kotlin/com/mongodb/jbplugin/inspections/AbstractMongoDbInspectionBridge.kt:97]: Looks good! I would however put this in the model, in a method, that encapsulates this logic, and then call the method here. A method like overwriteQueryTargetDatabase or by the likes.

File: packages/jetbrains-plugin/src/main/kotlin/com/mongodb/jbplugin/inspections/AbstractMongoDbInspectionBridge.kt (hunk 1 above)
Comment (himanshusinghs) [reply packages/jetbrains-plugin/src/main/kotlin/com/mongodb/jbplugin/inspections/AbstractMongoDbInspectionBridge.kt:37]: oo nice - I did not know that, thank you <3

File: packages/jetbrains-plugin/src/main/kotlin/com/mongodb/jbplugin/inspections/AbstractMongoDbInspectionBridge.kt (hunk 2 above)
Comment (himanshusinghs) [reply packages/jetbrains-plugin/src/main/kotlin/com/mongodb/jbplugin/inspections/AbstractMongoDbInspectionBridge.kt:97]: Actually part of it is what the query.copy is doing. I was trying to keep that a little generic for the purpose of aligning with the definition of copy method. Or do you mean to have this method (queryWithCollectionReference) entirely on the Node class?

Comment (himanshusinghs) [reply packages/jetbrains-plugin/src/main/kotlin/com/mongodb/jbplugin/inspections/AbstractMongoDbInspectionBridge.kt]: I tried something in [0276ed0](https://github.com/mongodb-js/intellij/pull/39/commits/0276ed0291e36242dbb6f65700db2701a90acfde). Was trying to keep the method pure hence keeping it static and still doing the copy

**Comment by kmruiz** [comment]
Don't forget to add the feature to the changelog :D

**Review by kmruiz — APPROVED** [review]

File: packages/jetbrains-plugin/src/main/kotlin/com/mongodb/jbplugin/inspections/bridge/FieldExistenceCheckInspectionBridge.kt
Comment (himanshusinghs) [inline packages/jetbrains-plugin/src/main/kotlin/com/mongodb/jbplugin/inspections/bridge/FieldExistenceCheckInspectionBridge.kt:1]: This looks deleted but I just renamed this file to JavaDriverFieldExistenceCheckInspectionBridge.kt

**Pull request merged**